pkg/server/api/agent/v1/service_test.go

```diff
@@ -1970,27 +1970,21 @@ func (s *serviceTest) setupNodes(ctx context.Context, t *testing.T) {
 }
 
 func (s *serviceTest) setupJoinTokens(ctx context.Context, t *testing.T) {
-	_, err := s.ds.CreateJoinToken(ctx, &datastore.CreateJoinTokenRequest{
-		JoinToken: &datastore.JoinToken{
-			Token:  "test_token",
-			Expiry: time.Now().Unix() + int64(60*10),
-		},
+	_, err := s.ds.CreateJoinToken(ctx, &datastore.JoinToken{
+		Token:  "test_token",
+		Expiry: time.Now().Unix() + int64(60*10),
```

Comment:
We should change the type on JoinToken.Expiry to `time.Time`.

Reply:
This proved problematic going to and from the database. 
time.Time.Unix() -> DB int64 and DB int64 -> time.Unix(model.expiry, 0) do not compare equally
time.Time.Unix() -> DB int64 and DB int64 -> time.Time{}.Add(time.Second*time.Duration(model.expiry)) do not compare equally
Is there another, perhaps idiomatic way, to do this?

Reply:
time.Now() has a monotonic clock while time.Unix(sec, 0) does not so the wall and ext values are different.
we could just use t.Unix() for all comparisons, although at that point I'd wonder at the value of having expiry be time.Time

Reply:
https://golang.org/pkg/time/#Time.Equal?

Reply:
Nope :-(

```
now := time.Now()
nowInt64 := now.Unix()
now2 := time.Unix(nowInt64, 0)
fmt.Println(now.Equal(now2))
fmt.Printf("now: %v\nnow2: %v\n", now, now2)
```
false
now: 2021-04-12 10:19:38.360396 -0700 PDT m=+0.000086398
now2: 2021-04-12 10:19:38 -0700 PDT


However, if we use a custom now() function to strip nanoseconds and the monotonic clock:

```
func NowSecondsUTC() time.Time {
	t := time.Now()
	return time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), 0, time.UTC)
}

func main() {
	now := NowSecondsUTC()
	nowInt64 := now.Unix()
	now2 := time.Unix(nowInt64, 0)
	fmt.Println(now.Equal(now2))
	fmt.Printf("now: %v\nnow2: %v\n", now, now2)
}
```

true
now: 2021-04-12 10:29:54 +0000 UTC
now2: 2021-04-12 03:29:54 -0700 PDT

Reply:
I think we're talking past each other. The use of `time.Now()` should be a test concern only. The granularity supported by both the API and the database is seconds (since Unix epoch). There should be no loss of precision moving to/from `int64` and `time.Time` via `time.Unix()` and `(*time.Time).Unix()`.
In the tests, instead of setting the token expiry to `time.Now()`, which will include nanosecond precision and is not a time that will be 1) received over the API, or 2) retrieved from the DB, we should set use `time.Now().Truncate(time.Second)`.